shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.mm

```diff
@@ -91,26 +91,18 @@ - (void)dealloc {
 
 @end
 
-@implementation DisplayLinkManager {
-  fml::scoped_nsobject<CADisplayLink> display_link_;
```

Comment:
You still need to hold an ivar to the display link to invalidate it later. The display link retains the target and without invalidation, the back reference to it will not be cleared.

Reply:
@chinmaygarde we release the display link https://github.com/flutter/engine/pull/22194/files#diff-ca0ba764dd3a3c6d10dfd05ea010d65973a4882b7f2549352367b97944a1fd8fR105, the reference to `DisplayLinkManager` would be nilled automatically.

Reply:
The reference to the DisplayLinkManager will only be nilled when the display link is unscheduled from the event loop right? That won't happen till you invalidate the link. So it is not entirely automatic.

Reply:
@chinmaygarde You're right, but we don't schedule the display link to runloop, and we also don't need invalidate it .